src/houghtransform.jl

```diff
@@ -1,84 +1,100 @@
+where r and t are parameters for normal form of line:
+    `x \* cos(t) + y \* sin(t) = r`
 
-r = length of perpendicular from (1,1) to the line
-t = angle between perpendicular from (1,1) to the line and x-axis
+-   `r` = length of perpendicular from (1,1) to the line
+-   `t` = angle between perpendicular from (1,1) to the line and x-axis
 
 The lines are generated by applying hough transform on the image.
 
 Parameters:
-    image       = Image to be transformed (eltype should be `Bool`)
-    ρ           = Discrete step size for perpendicular length of line
-    θ           = List of angles for which the transform is computed
-    threshold   = No of points to pass through line for considering it valid
-    linesMax    = Maximum no of lines to return
+-    `image`       = Image to be transformed (eltype should be `Bool`)
+-    `ρ`           = Discrete step size for perpendicular length of line
+-    `θ`           = List of angles for which the transform is computed
+-    `threshold`   = Accumulator threshold for line detection
+-    `linesMax`    = Maximum no of lines to return
+
+# Example
+```julia
+julia> img = load("line.jpg");
+
+julia> img_edges = canny(img, (Percentile(0.99), Percentile(0.97)), 1);
 
+julia> lines = hough_transform_standard(img_edges, 1, linspace(0,π,30), 40, 5)
+5-element Array{Tuple{Float64,Float64},1}:
+ (45.0,1.73329)  
+ (1.0,1.73329)   
+ (32.0,1.73329)  
+ (209.0,0.649985)
+ (-9.0,2.49161) 
+```
 """
 
 function hough_transform_standard{T<:Union{Bool,Gray{Bool}}}(
             img::AbstractArray{T,2},
-            ρ::Number, θ::Range,
+            ρ::Real, θ::Range,
             threshold::Integer, linesMax::Integer)
 
 
     #function to compute local maximum lines with values > threshold and return a vector containing them
-    function findlocalmaxima(accumulator_matrix::Array{Integer,2},threshold::Integer)
-        validLines = Vector{CartesianIndex}(0)
+    function findlocalmaxima!{T<:Integer}(validLines::AbstractVector{CartesianIndex{2}}, accumulator_matrix::Array{Int,2}, threshold::T)
         for val in CartesianRange(size(accumulator_matrix))
-            if  accumulator_matrix[val] > threshold &&
-                accumulator_matrix[val] > accumulator_matrix[val[1],val[2] - 1] &&
+            if  accumulator_matrix[val] >  threshold                             &&
+                accumulator_matrix[val] >  accumulator_matrix[val[1],val[2] - 1] &&
                 accumulator_matrix[val] >= accumulator_matrix[val[1],val[2] + 1] &&
-                accumulator_matrix[val] > accumulator_matrix[val[1] - 1,val[2]] &&
+                accumulator_matrix[val] >  accumulator_matrix[val[1] - 1,val[2]] &&
                 accumulator_matrix[val] >= accumulator_matrix[val[1] + 1,val[2]]
                 push!(validLines,val)
             end
         end
-        validLines
     end
 
     ρ > 0 || error("Discrete step size must be positive")
 
-    height, width = size(img)
+    height, width = indices(img)
```

Comment:
Might be slightly odd to continue to use these names if these are ranges rather than numbers.